src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs

```diff
@@ -63,38 +63,33 @@ public IEnumerable<string> Generate(string[] pinvokeModules, string outputPath)
         return signatures;
     }
 
-    private static bool HasAttribute(MemberInfo element, params string[] attributeNames)
+    private void EmitPInvokeTable(StreamWriter w, Dictionary<string, string> modules, List<PInvoke> pinvokes)
     {
-        foreach (CustomAttributeData cattr in CustomAttributeData.GetCustomAttributes(element))
-        {
-            try
+        w.WriteLine("// GENERATED FILE, DO NOT MODIFY");
+        w.WriteLine();
+
+        foreach (var pinvoke in pinvokes) {
+            // Handle special modules, and add them to the list of modules
+            // otherwise, skip them and throw an exception at runtime if they
+            // are called.
+            if (pinvoke.WasmLinkage && !modules.ContainsKey(pinvoke.Module))
```

Comment:
nit: Move `!modules.ContainsKey(pinvoke.Module)` to a separate if and don't repeat it (I wanted to submit a patch, but github says it's not supported here)